### ISSUE TITLE
Quadrature of discrete distribution can return points outside interval with rule "C"

### ISSUE THREAD

**Issue by fjansson** [jonathf/chaospy#304]
```
import chaospy as cp
seed = cp.DiscreteUniform(1, 2000)
cp.generate_quadrature(2, seed, rule="C")

(array([[   0, 1000, 2000]]), array([0.16666667, 0.66666667, 0.16666667]))
```
The first point, 0, is outside the interval of the distribution. This seems to happen when either bound is is +/- 1, with other bounds I have not seen it. With the rules G or discrete I haven't seen any problems.

**Comment by jonathf** [comment]
Yup, that is a bug. As noted in #303, the variables are continous underneath the hood. Here we have a rounding error at the edges that flips the other way. Fix added to version 4.2.1.

**Comment by fjansson** [comment]
Wow, that was quick! I confirm both this and #303 are fixed in 4.2.1. Thank you!
